### ISSUE TITLE
[BUG]: AttributeError: 'CatBoostClassifier' object has no attribute 'steps'

### ISSUE THREAD

**Issue by celestinoxp** [pycaret/pycaret#2769]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

Using code to predict soccer, other algorithms works... but if compare_models choose catboost algo... in finalize_model shows:

AttributeError: 'CatBoostClassifier' object has no attribute 'steps'

### Reproducible Example

```python
private code, sorry
```

### Expected Behavior

no errors

### Actual Results

```python-traceback
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [13], in <cell line: 1>()
----> 1 final_model = finalize_model(tuned_model)

File ~\anaconda3\envs\teste2\lib\site-packages\pycaret\internal\utils.py:922, in check_if_global_is_not_none.<locals>.decorator.<locals>.wrapper(*args, **kwargs)
    920     if globals_d[name] is None:
    921         raise ValueError(message)
--> 922 return func(*args, **kwargs)

File ~\anaconda3\envs\teste2\lib\site-packages\pycaret\classification\functional.py:2228, in finalize_model(estimator, fit_kwargs, groups, model_only, experiment_custom_tags)
   2175 @check_if_global_is_not_none(globals(), _CURRENT_EXPERIMENT_DECORATOR_DICT)
   2176 def finalize_model(
   2177     estimator,
   (...)
   2181     experiment_custom_tags: Optional[Dict[str, Any]] = None,
   2182 ) -> Any:
   2184     """
   2185     This function trains a given estimator on the entire dataset including the
   2186     holdout set.
   (...)
   2225 
   2226     """
-> 2228     return _CURRENT_EXPERIMENT.finalize_model(
   2229         estimator=estimator,
   2230         fit_kwargs=fit_kwargs,
   2231         groups=groups,
   2232         model_only=model_only,
   2233         experiment_custom_tags=experiment_custom_tags,
   2234     )

File ~\anaconda3\envs\teste2\lib\site-packages\pycaret\classification\oop.py:2849, in ClassificationExperiment.finalize_model(self, estimator, fit_kwargs, groups, model_only, experiment_custom_tags)
   2796 def finalize_model(
   2797     self,
   2798     estimator,
   (...)
   2802     experiment_custom_tags: Optional[Dict[str, Any]] = None,
   2803 ) -> Any:
   2805     """
   2806     This function trains a given estimator on the entire dataset including the
   2807     holdout set.
   (...)
   2846 
   2847     """
-> 2849     return super().finalize_model(
   2850         estimator=estimator,
   2851         fit_kwargs=fit_kwargs,
   2852         groups=groups,
   2853         model_only=model_only,
   2854         experiment_custom_tags=experiment_custom_tags,
   2855     )

File ~\anaconda3\envs\teste2\lib\site-packages\pycaret\internal\pycaret_experiment\supervised_experiment.py:4731, in _SupervisedExperiment.finalize_model(self, estimator, fit_kwargs, groups, model_only, experiment_custom_tags)
   4716 pipeline_final, model_fit_time = self._create_model(
   4717     estimator=estimator,
   4718     cross_validation=False,
   (...)
   4727     model_only=False,
   4728 )
   4730 if self._ml_usecase != MLUsecase.TIME_SERIES:
-> 4731     self.predict_model(pipeline_final, data=self.test, verbose=False)
   4732 else:
   4733     self.predict_model(pipeline_final, X=self.X_test, verbose=False)

File ~\anaconda3\envs\teste2\lib\site-packages\pycaret\classification\oop.py:2785, in ClassificationExperiment.predict_model(self, estimator, data, probability_threshold, encoded_labels, raw_score, drift_report, round, verbose)
   2706 def predict_model(
   2707     self,
   2708     estimator,
   (...)
   2715     verbose: bool = True,
   2716 ) -> pd.DataFrame:
   2718     """
   2719     This function predicts ``Label`` and ``Score`` (probability of predicted
   2720     class) using a trained model. When ``data`` is None, it predicts label and
   (...)
   2782 
   2783     """
-> 2785     return super().predict_model(
   2786         estimator=estimator,
   2787         data=data,
   2788         probability_threshold=probability_threshold,
   2789         encoded_labels=encoded_labels,
   2790         raw_score=raw_score,
   2791         drift_report=drift_report,
   2792         round=round,
   2793         verbose=verbose,
   2794     )

File ~\anaconda3\envs\teste2\lib\site-packages\pycaret\internal\pycaret_experiment\supervised_experiment.py:4922, in _SupervisedExperiment.predict_model(self, estimator, data, probability_threshold, encoded_labels, raw_score, drift_report, round, verbose, ml_usecase, preprocess)
   4920 pipeline = deepcopy(estimator)
   4921 # Temporarily remove final estimator so it's not used for transform
-> 4922 final_step = pipeline.steps[-1]
   4923 estimator = final_step[-1]
   4924 pipeline.steps = pipeline.steps[:-1]

AttributeError: 'CatBoostClassifier' object has no attribute 'steps'
```

### Installed Versions

<details>
pycaret rc3
catboost 1.0.6
</details>

**Comment by tvdboom** [comment]
We can't fix the issue without code to reproduce it. If you have private data, please reproduce the error on a public dataset and share it here.

**Comment by jnsflt** [comment]
Hey @tvdboom! I experienced the same issue today. Here is code for you to reproduce it:

`
from pycaret.datasets import get_data
df = get_data('titanic')
exp = ClassificationExperiment()
exp.setup(data=df, target='Survived')
clf = exp.create_model('catboost')

path = '.../catboost_clf'
exp.save_model(clf, path)
model = exp.load_model(path)
exp.predict_model(estimator=model, data=df)
`

**Comment by jnsflt** [comment]
This only happens if you previously save the model.

**Comment by tvdboom** [comment]
This error seems to appear only for the catboost model. I pushed a fix in #2841. Until the merge, try any other model.
